charts/portworx/templates/storage-cluster.yaml

```diff
@@ -1,193 +1,84 @@
-    "{{ required "A valid ETCD url in the format etcd:http://<your-etcd-endpoint> is required. Verify that the key is correct and there isnt any typo in specifying that, also ensure it is accessible from all node of your kubernetes cluster" .etcdEndPoint}}"
-    {{- else }}
+    internal: {{ .Values.kvdb.internal }}
+    {{- with .Values.etcd.endpoints }}
     endpoints:
-      {{- $endpoints := $etcdEndPoints | split ";" }}
-      {{- range $key, $val := $endpoints }}
-      - {{$val}}
-      {{- end }}
+      {{- toYaml . | nindent 6 }}
     {{- end }}
-    {{- if ne $kvdbAuthSecret "none" }}
     authSecret: {{ .Values.kvdb.authSecretName }}
-    {{- end }}
-  {{- end }}
-
-{{- if and (ne $drives "none") (not (hasPrefix "/" $drives)) }}
-  cloudStorage:
-    deviceSpecs:
-      {{- $driveNames := $drives | split ";" }}
-      {{- range $index, $name := $driveNames }}
-      - {{ $name }}
-      {{- end }}
-    {{- if and (ne $kvdbDevice "none") (not (hasPrefix "/" $kvdbDevice)) }}
-    kvdbDeviceSpec: {{ $kvdbDevice }}
-    {{- end }}
-    {{- if and (ne $journalDevice "none") (not (hasPrefix "/" $journalDevice)) }}
-    journalDeviceSpec: {{ $journalDevice }}
-    {{- end }}
-    {{- if ne (.Values.maxStorageNodesPerZone | int) 0 }}
-    maxStorageNodesPerZone: {{ .Values.maxStorageNodesPerZone }}
-    {{- end }}
-{{- else }}
-  {{- if $AKSInstall }}
+  {{- with .Values.cloudStorage }}
   cloudStorage:
-    deviceSpecs:
-      - type=Premium_LRS,size=150
-    {{- if and (ne $kvdbDevice "none") (not (hasPrefix "/" $kvdbDevice)) }}
-    kvdbDeviceSpec: {{ $kvdbDevice }}
-    {{- end }}
-    {{- if and (ne $journalDevice "none") (not (hasPrefix "/" $journalDevice)) }}
-    journalDeviceSpec: {{ $journalDevice }}
-    {{- end }}
-    {{- if gt (.Values.maxStorageNodesPerZone | int) 0 }}
-    maxStorageNodesPerZone: {{ .Values.maxStorageNodesPerZone }}
-    {{- end }}
-  {{- else }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.storage }}
   storage:
-    {{- if hasPrefix "/" $drives }}
-    devices:
-      {{- $driveNames := $drives | split ";" }}
-        {{- range $index, $name := $driveNames }}
-        - {{ $name }}
-        {{- end }}
-    {{- end }}
-    {{- if or $usefileSystemDrive $deployEnvironmentIKS }}
-    forceUseDisks: true
-    {{- end }}
-    {{- if eq $usedrivesAndPartitions true }}
-    useAllWithPartitions: true
-    {{- else }}
-    useAll: true
-    {{- end }}
-    {{- if and (ne $kvdbDevice "none") (hasPrefix "/" $kvdbDevice) }}
-    kvdbDevice: {{ $kvdbDevice }}
-    {{- end }}
-    {{- if and (ne $journalDevice "none") (or (hasPrefix "/" $journalDevice) (eq "auto" $journalDevice)) }}
-    journalDevice: {{ $journalDevice }}
-    {{- end }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
-{{- end }}
-
-  {{- if or (ne $dataInterface "none") (ne $managementInterface "none") }}
+    kvdbDeviceSpec: {{ .Values.kvdb.device }}
+    journalDeviceSpec: {{ .Values.journalDevice }}
```

Comment:
These should only be present when the user passes in the value, else they should not exist in the spec.